dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/calculate-fixed-fee.ts

```diff
@@ -0,0 +1,76 @@
+import Big from 'big.js';
+import { differenceInDays, isBefore, startOfMonth } from 'date-fns';
+import orderBy from 'lodash.orderby';
+import { MonthAndYear, ReportPeriod, TfmFacilityAmendment } from '@ukef/dtfs2-common';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { NotFoundError } from '../../../../../errors';
+import { convertTimestampToDate } from '../../../../../helpers';
+
+const getLatestCompletedAmendmentCoverEndDate = (amendments: TfmFacilityAmendment[]): number | undefined => {
+  const completedAmendments = amendments.filter(({ status }) => status === 'Completed');
+  const latestAmendmentWithDefinedCoverEndDate = orderBy(completedAmendments, ['updatedAt'], ['desc']).find(({ coverEndDate }) => coverEndDate);
+  return latestAmendmentWithDefinedCoverEndDate?.coverEndDate ?? undefined;
+};
+
+const getLatestTfmFacilityValues = async (
+  facilityId: string,
+): Promise<{
+  coverEndDate: Date;
+  coverStartDate: Date;
+  dayCountBasis: number;
+  interestPercentage: number;
+}> => {
+  const tfmFacility = await TfmFacilitiesRepo.findOneByUkefFacilityId(facilityId);
+  if (!tfmFacility) {
+    throw new NotFoundError(`Failed to find a tfm facility with ukef facility id '${facilityId}'`);
+  }
+
+  const { coverEndDate: snapshotCoverEndDate, coverStartDate, dayCountBasis, interestPercentage } = tfmFacility.facilitySnapshot;
+  const latestAmendedCoverEndDate = tfmFacility.amendments ? getLatestCompletedAmendmentCoverEndDate(tfmFacility.amendments) : undefined;
+
+  const coverEndDate = latestAmendedCoverEndDate ?? snapshotCoverEndDate;
+  if (!coverEndDate) {
+    throw new NotFoundError(`Failed to find a cover end date for the tfm facility with ukef facility id '${facilityId}`);
+  }
+
+  if (!coverStartDate) {
+    throw new NotFoundError(`Failed to find a cover start date for the tfm facility with ukef facility id '${facilityId}`);
+  }
+
+  return {
+    coverEndDate: convertTimestampToDate(coverEndDate),
+    coverStartDate: convertTimestampToDate(coverStartDate),
+    dayCountBasis,
+    interestPercentage: new Big(interestPercentage).div(100).toNumber(),
+  };
+};
+
+const getNumberOfDaysRemainingInCoverPeriod = (reportPeriodStart: MonthAndYear, coverStartDate: Date, coverEndDate: Date): number => {
+  const currentReportPeriodStartMonthStart = startOfMonth(new Date(`${reportPeriodStart.year}-${reportPeriodStart.month}`));
```

Comment:
Could we use the `getDateFromMonthAndYear` helper inside the startOfMonth call here?